filebeat/input/v2/error.go

```diff
@@ -0,0 +1,105 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package v2
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// LoadError is returned by Loaders in case of failures.
+type LoadError struct {
+	// Name of input/module that failed to load (if applicable)
+	Name string
+
+	// Reason why the loader failed. Can either be the cause reported by the
+	// Plugin or some other indicator like ErrUnknown
+	Reason error
+
+	// (optional) Message to report in additon.
+	Message string
+}
+
+// SetupError indicates that the loader initialization has detected
+// errors in individual plugin configurations or duplicates.
+type SetupError struct {
+	Fails []error
+}
+
+// ErrUnknown indicates that the plugin type does not exist. Either
+// because the 'type' setting name does not match the loaders expectations,
+// or because the type is unknown.
+var ErrUnknown = errors.New("unknown input type")
```

Comment:
Do you mind renaming it to `ErrUnknownInput`? This name is a bit misleading to me. It might imply that the error is unknown.

Reply:
Good point, will change.